inference-engine/src/transformations/src/transformations/convert_precision.cpp

```diff
@@ -57,39 +118,42 @@ bool ngraph::pass::ConvertPrecision::run_on_function(std::shared_ptr<ngraph::Fun
         }
     };
 
-    register_constants(f);
-
-    auto convert_node_precision = [this, &const_to_internal_output](std::shared_ptr<Node> & node) {
-        // As input type could changed we need to propagate output type calculation manually
-        node->validate_and_infer_types();
-
+    auto convert_node_output_precision = [this, &const_to_internal_output](std::shared_ptr<Node> & node) {
         for (auto output : node->outputs()) {
             if (output.get_element_type() == m_from) {
                 // Handle case with Constants as they can have consumers from other nGraph Function object
                 if (ngraph::op::is_constant(node) && const_to_internal_output.count(node)) {
                     fuse_type_to_constant(node, m_to, const_to_internal_output.at(node));
-                    continue;
+                    break;
                 }
 
                 // If node type in map and convert can be fused into node we skip Convert creation
                 if (type_to_fuse.count(node->get_type_info()) &&
                     type_to_fuse.at(node->get_type_info())(node, m_to, output.get_index())) {
```

Comment:
Check if node is not TypeRelaxed already. Just dynamic_cast to TypeRelaxedBase. If it is already TypeRelaxedBased, use corresponding methods to set desired types instead of constructing new TypeRelaxed which will probably fail at some point (because you lose old settings for types in original TypeRelaxed).

Reply:
Probably, it is better idea to add this check to TypeRelaxed ctor and behave differently, but it is not universal anyway (work in case if you pass original object to the ctor, but not separate arguments).

Reply:
Done.